File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, bool> m1;
        
        for(int i=0;i<nums.size();i++){
            if(m1.find(nums[i])!=m1.end())
                return true;
            m1[nums[i]] = true;
        }
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> s1 = new HashSet<Integer>();

        for(int n: nums){
            if(s1.contains(n)){
                return true;
            }
            s1.add(n);
        }
        return false;
    }
}